erpnext/crm/doctype/appointment/appointment.js

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2019, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Appointment', {
+	refresh: function(frm) {
+		if(frm.doc.lead){
+			frm.add_custom_button(__(frm.doc.lead),()=>{
```

Comment:
```suggestion
			frm.add_custom_button(frm.doc.lead,()=>{
```

The button name need not be translated in this case, since `frm.doc.lead` will be entered by user anyway